src/main/java/racingcar/domain/Cars.java

```diff
@@ -0,0 +1,63 @@
+package racingcar.domain;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public class Cars {
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        this.cars = Collections.unmodifiableList(cars);
+    }
+
+    protected int maxPosition() {
+        int mxPosition = 0;
+        for (Car car : cars) {
+            mxPosition = car.max(mxPosition);
+        }
+        return mxPosition;
+    }
+
+    protected void moveAll() {
+        for (Car car : cars) {
+            car.move(new RandomNumber());
+        }
+    }
+
+    protected Winners filterWinners() {
+        int maxPosition = maxPosition();
+
+        Winners winners = new Winners();
+        for (Car car : cars) {
+            winners.addWinner(maxPosition, car);
+        }
+
+        return winners;
```

Comment:
여기서 위너를 구할때 아래 같이 바꿔볼수 있을 것 같아요. 그럼 굳이 `winners.addWinner` 를 따로 구현할 필요는 없겠죠? 😄
지난번에도  스트림을 언급했는데, 스트림을 쓰면 여러 작업을 chaining 해서 쓸수 있기 때문에 분명 장점이 있어요.
그러나 맹목적으로 stream을 쓰기 보단 어떤 경우는 일반 for문이 더 빠르기도 하니 적절히 혼용해서 쓰시길 추천합니다. 

```suggestion
        List<Car> winnerCars = cars.stream()
                .filter(car -> car.isWinner(maxPosition()))
                .collect(Collectors.toUnmodifiableList());

        return new Winners(winnerCars);

Reply:
네 좀 더 익숙해지도록 노력하겠습니다.
감사합니다